docs/advanced.rst

```diff
@@ -562,3 +562,79 @@ or the record invocation: ``app="SomeAppName"``.
+
+    [appName]_[modelName]
+
+where the history models are created in a dedicated app using the "app" option thus making the resultant history table
+name much more aligned with the base table it is tracking
+
+To configure history models to use a different name for the history model class, use an option named ``custom_model_name``
+
+The value for this option can be a string or a callable.
+
+The most simple use case is illustrated below:
+
+.. code-block:: python
+
+    class ModelNameExample(models.Model):
+        history = HistoricalRecords(
+            custom_model_name='SimpleHistoricalModelNameExample'
+        )
+
+As an example using the callable mechanism, the below changes the default prefix `Historical` to `Audit`:
+
+.. code-block:: python
+
+    class Poll(models.Model):
+        question = models.CharField(max_length=200)
+        history = HistoricalRecords(custom_model_name=lambda x:f'Audit{x}')
+
+    class Opinion(models.Model):
+        opinion = models.CharField(max_length=2000)
+
+    register(Opinion, custom_model_name_prefix=lambda x:f'Audit{x}')
+    
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+If the app the models are defined in is `yoda` then the corresponding history table names would be `yoda_auditpoll` and `yoda_auditopinion`
+ 
+You could also use a base class for your models and declare the history instantiation in the base class.
+All models that inherit the base model would automatically get this custom name.
+Using this mechanism requires using the `inherit` option.
+
+.. code-block:: python
+
+    class BaseModel(models.Model):
+        history = HistoricalRecords(inherit=True, app="yoda", custom_model_name_prefix=lambda x:f'Audit{x}')
+        class Meta:
+            abstract = True
+
+    class Poll(BaseModel):
+        question = models.CharField(max_length=200)
+
+    class Opinion(BaseModel):
+        opinion = models.CharField(max_length=2000)
+
+The resulting history class names would be `AuditPoll` and `AuditOpinion`.
+The history model tables will be named `yoda_auditpoll` and `yoda_auditopinion`.
+
+If the instantiation in the above example used `lambda x:f'{x}'` for the `custom_model_name` option
+  then the resulting history class names would be `HistoricalPoll` and `HistoricalOpinion`.
+  and the corresponding history table names would be `yoda_poll` and `yoda_opinion`
+ 
+ NOTE: the class name of the historical class in this situation would be `HistoricalPoll` and NOT just `Poll`
+       to ensure it is different to the tracked class since using the `inherit` option that is necessary when using
+       an abstract class forces the historical model to be created in the same module as the model class the history model
+       is created from and would otherwise have the identical name.
+ 
+IMPORTANT: Setting `custom_model_name` to `lambda x:f'{x}'` requires that the history tables
```

Comment:
Why?

Reply:
The resulting history class name for history on model "models.Poll" would be "models.Poll" because of  "inherit=True". Though clearly it is a diffferent instance of the class I was not sure if that could somehow cause issues so went the safe route and ensured the history class is different. If you are happy with them being the same I can change it.

Reply:
Ah got it. What's the functionality if the custom_model_name is a string that's the same as the model_name? Does it error out? Don't think you need to note this, just have it throw an error in that case.

Reply:
It only generates a warning if the "custom_model_name" is a string with the same name as the class it will track (same 'app' and inherit=False):
`RuntimeWarning: Model 'xxx.Yyy' was already registered. Reloading models is not advised as it can lead to inconsistencies, most notably with related models`

So is not caught within history app. Want me to check for it in the same way as done for callable mechanism?

Reply:
It doesn't generate the warning if the `custom_model_name` is a callable?

Reply:
No because the code checks for this and prepends "Historical" to the class name:
`    def get_history_model_name(self, model, attrs):
        if self.custom_model_name:
            if callable(self.custom_model_name):
                name = self.custom_model_name(model._meta.object_name)
                if not name.lower() == model._meta.object_name.lower():
                    return name
                self.check_custom_model_name_unique(model)
                # Override table_name and make class name use the standard option
                # because otherwise it creates the same class as the concrete class
                attrs["Meta"].db_table = self.app + "_" + name.lower()
            else:
                # Treat as a simple string
                return self.custom_model_name
        return "Historical{}".format(model._meta.object_name)
`

Reply:
Actually it only prepends "Historical" if the "app" names are different. 
If the app names are the same it  raises and error ... so can make it do the same for string based "custom_model_name"